### DISCUSSION TITLE
Use `listen_event` from a SimpleConnection

### DISCUSSION THREAD

**Discussion by tombowditch** [fernandobatels/rsfbclient#148] [Q&A]
Hello,

I am converting my `Connection<NativeFbClient<DynLoad>>` into a `SimpleConnection` so I can wrap it in a internal db client. I'm trying to now use events; but unable to.

`.listen_event` is available on `Connection<NativeFbClient<DynLoad>>` (`conn1` below) but seemingly not on the same `SimpleClient` (`c1` below):

![CleanShot 2023-10-30 at 17 47 20](https://github.com/fernandobatels/rsfbclient/assets/5332144/36df42ff-fd65-405f-81c1-5b8c6ecc50c7)

Is this expected?

Full code:

```rust
        let conn = rsfbclient::builder_native()
            .with_dyn_load(
                "/Library/Frameworks/Firebird.framework/Versions/A/Libraries/libfbclient.dylib",
            )
            .with_remote()
            .db_name("xxxxxxxxxxxxx")
            .host("xxxxxxxxxx")
            .transaction(TransactionConfiguration {
                lock_resolution: TrLockResolution::NoWait,
                ..TransactionConfiguration::default()
            })
            .connect()?;

        conn.listen_event("HELLO".into(), |c| Ok(true));

        let c1: SimpleConnection = conn.into();
        c1.listen_event("HELLO1", |c| Ok(true)); // <-- errors

```

**Comment by fernandobatels** [comment]
I think this is an error on `SimpleConnection` implementation, I will fix

**Reply by tombowditch** [reply]
Excellent thanks @fernandobatels 

For what it's worth, I found a workaround by adding `rsfbclient-native` into my project too - that exports `NativeFbClient` publicly so I am able to use `.try_into()` on a `SimpleConnection` to convert it back into a `Connection<NativeFbClient<DynLoad>>`:

![image](https://github.com/fernandobatels/rsfbclient/assets/5332144/23ccb20c-5325-4f96-befe-716131ba179e)
(you can see `conn_back` here is made from `SimpleConnection`, and I'm able to call `.listen_event` on it 👍)

**Reply by fernandobatels** [reply]
Nice workaround hehe
